exporters/committime/app.py

```diff
@@ -22,233 +18,31 @@
-                    commit_sha, metric.commit_timestamp))
-
-            return metric
-
-        except Exception as e:
-            logging.warning("Build %s/%s in app %s is missing required attributes to collect data. Skipping."
-                            % (namespace, build.metadata.name, app))
-            logging.debug(e, exc_info=True)
-            return None
-
-
-class CommitTimeMetric:
-    _prefix_pattern = "https://%s/repos/"
-    _defaultapi = "api.github.com"
-    _prefix = _prefix_pattern % _defaultapi
-    _suffix = "/commits/"
-
-    def __init__(self, app_name, _gitapi=None):
-        self.name = app_name
-        self.labels = None
-        self.repo_url = None
-        self.commiter = None
-        self.commit_hash = None
-        self.commit_time = None
-        self.commit_timestamp = None
-        self.build_name = None
-        self.build_config_name = None
-        self.image_location = None
-        self.image_name = None
-        self.image_tag = None
-        self.image_hash = None
-        if _gitapi is not None and len(_gitapi) > 0:
-            logging.info("Using non-default GitHub API: %s" % (_gitapi))
-            self._prefix = self._prefix_pattern % _gitapi
-
-    def getCommitTime(self):
-        myurl = self.repo_url
-        url_tokens = myurl.split("/")
-        url = self._prefix + url_tokens[3] + "/" + url_tokens[4].split(".")[0] + self._suffix + self.commit_hash
-        response = requests.get(url, auth=(username, token))
-        if response.status_code != 200:
-            # This will occur when trying to make an API call to non-Github
-            logging.warning("Unable to retrieve commit time for build: %s, hash: %s, url: %s. Got http code: %s" % (
-                self.build_name, self.commit_hash, url_tokens[2], str(response.status_code)))
-        else:
-            commit = response.json()
-            try:
-                self.commit_time = commit['commit']['committer']['date']
-                self.commit_timestamp = pelorus.convert_date_time_to_timestamp(self.commit_time)
-            except Exception:
-                logging.error("Failed processing commit time for build %s" % self.build_name, exc_info=True)
-                logging.debug(commit)
-                raise
+class GitFactory:
+    @staticmethod
+    def getCollector(username, token, namespaces, apps, git_api, git_type):
+        if git_type == "gitlab":
+            return GitLabCommitCollector("", "", "", "")
+        if git_type == "github":
+            return GitHubCommitCollector(username, token, namespaces, apps, git_api)
+        if git_type == "bitbucket":
+            return BitbucketCommitCollector("", "", "", "")
 
 
 if __name__ == "__main__":
     pelorus.check_required_config(REQUIRED_CONFIG)
     username = os.environ.get('GITHUB_USER')
     token = os.environ.get('GITHUB_TOKEN')
     git_api = os.environ.get('GITHUB_API')
+    git_type = os.environ.get('GIT_TYPE')
```

Comment:
I think we should change `git_type` to `git_provider`.

Reply:
I think we should change them all to be `GIT_xxx`.  The only caveat to that is what if I want to run two committime exporters against two different Git providers.

Reply:
Agreed with @themoosman that we should rename all these ENV vars.  Making the change now will break existing deployments and we will need a strategy for managing separately running instances.  

Should we make this change now with the refactor, or leave the 3 existing ENV vars and open an issue to fix?

Reply:
Ok I think I fixed it